MapPointMapper/ViewController.swift

```diff
@@ -108,14 +109,31 @@ class ViewController: NSViewController, MKMapViewDelegate, NSTextFieldDelegate {
     renderer.strokeColor = colorWell.color
     return renderer
   }
-  
-  // MARK: NSTextFieldDelegate
-  override func keyUp(theEvent: NSEvent) {
-    if theEvent.keyCode == 36 { // 36 is the return key apparently
-      addLineFromTextPressed(self.addLineFromTextButton)
+
+  @IBAction func searchForLocation(sender: NSObject) {
+    if searchfield.stringValue.isEmpty { return }
+    renderLocationSearch(searchfield.stringValue as NSString)
+    searchfield.stringValue = ""
+  }
+
+
+  private func renderLocationSearch(input: NSString) {
+    let geocoder = CLGeocoder()
+    geocoder.geocodeAddressString(input as String) {
+      if let placemarks = $0 {
+        let placemark = placemarks.first! as CLPlacemark
```

Comment:
What happens if we get back an empty set? This'll go :boom:. Probably want to `guard let place mark = placemarks.first` Could probably do something like

```swift
private func renderLocationSearch(input: NSString) {
    let geocoder = CLGeocoder()
    geocoder.geocodeAddressString(input as String) { (placemarks, error) in
      guard let placemark = placemarks?.first else {
        print(error ?? "")
        return
      }
      let region = MKCoordinateRegion(center: placemark.location?.coordinate ?? CLLocationCoordinate2D(),
                                      span: MKCoordinateSpan(latitudeDelta: 0.8, longitudeDelta: 0.8))
      self.mapview.setRegion(region, animated: true)
    }
  }
```